pandas/core/window.py

```diff
@@ -634,7 +634,7 @@ def validate(self):
         super().validate()
 
         window = self.window
-        if isinstance(window, (list, tuple, np.ndarray, libwindow_custom.BaseIndexer)):
+        if isinstance(window, (list, tuple, np.ndarray, BaseIndexer)):
```

Comment:
can we make this a module level method: is_window_indexer?

Reply:
then add a to_window_indexer function (which takes the same args and returns a BaseIndexer subclass)

Reply:
This check specifically is if users pass a custom BaseIndexer subclass, but yes it could be a `is_window_indexer` check.

Not totally clear on a `to_window_indexer` utility?

Reply:
I mean the conversion from something that passes ``is_window_indexer()`` will convert to an BaseIndexer (or subclass). It just makes the code simpler as these can be defined in pandas/core/window/indexer.py (see comment above). We can do this refactor later i think.

Reply:
So in the context of this check, this is just in the `validate` stage (validate input arguments) i.e. I don't think any conversions should happen _here_ necessarily.

I do think a `is_window_indexer` may be useful at a later stage.